.github/maintainers_guide.md

```diff
@@ -48,13 +48,15 @@ Releasing can feel intimidating at first, but rest assured: if you make a mistak
 
     - If issues are still open, discuss with the team about whether the open issues should be moved to a future release or if the release should be held off until the issues are resolved.
 
-2. Make sure your local `main` branch has the latest changes.
+    - Take a look at all issues under the Milestone to make sure that the type of issues included aligns with the Milestone name based on [semantic versioning](https://semver.org/). If the issues do not align with the naming of the Milestone (ex: if the issues are all bug fixes, but the Milestone is labeled as a minor release), then you can tweak the Milestone name to reflect the correct versioning.
+
+2. Make sure your local `main` branch has the latest changes. Then, open a new branch off of your local `main` branch for the release.
```

Comment:
Discussion point here: should we come up with a naming convention for this release branch?